### ISSUE TITLE
False positive `pointless-statement` for `:=` operator

### ISSUE THREAD

**Issue by clavedeluna** [PyCQA/pylint#7760]
### Bug description

Issue discovered in [this PR](https://github.com/PyCQA/pylint/pull/7752#discussion_r1020904341)

```
# test.py
(walrus := False)
print(walrus)
```

You can also see this by running 

`pylint tests/functional/s/superfluous_parens_walrus_py38.py --enable=all`

```
...
tests/functional/s/superfluous_parens_walrus_py38.py:72:0: I0020: Suppressed 'pointless-statement' (from line 2) (suppressed-message)
tests/functional/s/superfluous_parens_walrus_py38.py:73:0: I0020: Suppressed 'pointless-statement' (from line 2) (suppressed-message)
tests/functional/s/superfluous_parens_walrus_py38.py:69:0: I0020: Suppressed 'pointless-statement' (from line 2) (suppressed-message)
tests/functional/s/superfluous_parens_walrus_py38.py:70:0: I0020: Suppressed 'pointless-statement' (from line 2) (suppressed-message)
```
Once this issue is fixed, please go back and remove ` pointless-statement` in pylint-disable in file tests/functional/s/superfluous_parens_walrus_py38.py

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:1:0: W0104: Statement seems to have no effect (pointless-statement)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```
```


### Expected behavior

No `pointless-statement` message in the output.

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.12
Python 3.11.0 (main, Oct 25 2022, 14:13:24) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

macOS

**Comment by cdce8p** [comment]
`(walrus := False)` might technically not be a `pointless-statement`, it's bad code never the less IMO.
If we remove the warning, we should add a new one for it.

**Comment by cdce8p** [comment]
Opened #7763 which would add a new `named-expr-without-context` check.
